docling/backend/html_backend.py

```diff
@@ -99,490 +83,242 @@ def supported_formats(cls) -> set[InputFormat]:
+        caption_item = None
+        if alt:
+            caption_item = doc.add_text(DocItemLabel.CAPTION, alt, parent=parent)
+
+        src_url = img_tag.get("src", "")
+        width = img_tag.get("width", str(DEFAULT_IMAGE_WIDTH))
+        height = img_tag.get("height", str(DEFAULT_IMAGE_HEIGHT))
+        img_ref: Optional[ImageRef] = None
+
+        if self.image_options == ImageOptions.EMBEDDED:
+            try:
+                if src_url.startswith("http"):
+                    img = Image.open(requests.get(src_url, stream=True).raw)
+                elif src_url.startswith("data:"):
+                    data = re.sub(r"^data:image/.+;base64,", "", src_url)
+                    img = Image.open(BytesIO(base64.b64decode(data)))
+                else:
+                    return
+                img_ref = ImageRef.from_pil(img, dpi=int(img.info.get("dpi", (72,))[0]))
+            except (FileNotFoundError, UnidentifiedImageError) as e:
+                _log.warning(f"Could not load image (src={src_url}): {e}")
+                return
+
+        elif self.image_options == ImageOptions.REFERENCED:
+            try:
+                img_ref = ImageRef(
+                    uri=AnyUrl(src_url),
+                    dpi=72,
+                    mimetype="image/png",
+                    size=Size(width=float(width), height=float(height)),
                 )
-                if row_header:
-                    row_span -= 1
-                while (
-                    col_idx < num_cols
-                    and grid[row_idx + start_row_span][col_idx] is not None
-                ):
-                    col_idx += 1
-                for r in range(start_row_span, start_row_span + row_span):
-                    for c in range(col_span):
-                        if row_idx + r < num_rows and col_idx + c < num_cols:
-                            grid[row_idx + r][col_idx + c] = text
-
-                table_cell = TableCell(
-                    text=text,
-                    row_span=row_span,
-                    col_span=col_span,
-                    start_row_offset_idx=start_row_span + row_idx,
-                    end_row_offset_idx=start_row_span + row_idx + row_span,
-                    start_col_offset_idx=col_idx,
-                    end_col_offset_idx=col_idx + col_span,
-                    column_header=col_header,
-                    row_header=((not col_header) and html_cell.name == "th"),
+            except ValidationError as e:
+                _log.warning(f"Could not load image (src={src_url}): {e}")
+                return
+
+        doc.add_picture(image=img_ref, caption=caption_item, parent=parent)
+
+    def _parse_table(self, table_tag: Tag, doc: DoclingDocument, parent) -> TableData:
+        rows = []
+        for sec in ("thead", "tbody", "tfoot"):
+            section = table_tag.find(sec)
+            if section:
+                rows.extend(section.find_all("tr", recursive=False))
```

Comment:
Great! Glad to see `recursive=False`, which will help handle nested tables more effectively instead of the current behavior of dropping the whole table. #1613